### PR TITLE
Implemented Query

### PR DESCRIPTION
**Pull request by Retriever5500**
Implemented all necessary query methods. Should we encounter a case where we'd need more query method, the addition shall be rather easy,
@Navid-Ebadi-2003 @MMDPROJECT please review the code.

### CODE REVIEW

File: src/main/java/Server/Database/Query.java [hunk 1]
```diff
@@ -1,6 +1,753 @@
...
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return true;
+        }  
+
+    }
+
+    public static synchronized void signUpUser(UUID userId, String username, String email, String password) {
+        final String query = """
+                INSERT INTO User values(?, ?, ?, ?, ?, ?, ?, ?, ?);
```
Comment (MMDPROJECT) [inline src/main/java/Server/Database/Query.java]: Please consider deleting extra `?` after merging `database` into `dev`

File: src/main/java/Server/Database/Query.java [hunk 2]
```diff
@@ -1,6 +1,753 @@
...
+                """; 
+
+        try {
+            PreparedStatement pstmt = connection.prepareStatement(query);
+
+            pstmt.setString(1, userId.toString());
+            pstmt.setString(2, username);
+            pstmt.setString(3, email);
+            pstmt.setString(4, password);
```
Comment (MMDPROJECT) [inline src/main/java/Server/Database/Query.java:83]: `password` is hashed before we call `signUpUser` in `Service`?

File: src/main/java/Server/Database/Query.java [hunk 3]
```diff
@@ -1,6 +1,753 @@
...
+
+        try {
+            PreparedStatement pstmt = connection.prepareStatement(query);
+            // Setting the username value into the query
+            pstmt.setString(1, username);
+            ResultSet rs = pstmt.executeQuery();
+            if(rs.next()) {
+                String storedPass = rs.getString("password");
+                UUID userid = UUID.fromString(rs.getString("user_id"));
+                // TODO: hash the password
```
Comment (MMDPROJECT) [inline src/main/java/Server/Database/Query.java:140]: [In here please specify that where do we usually hash the password](https://github.com/Navid-Ebadi-2003/spotify_project/pull/14/files#r1222657812)

File: src/main/java/Server/Database/Query.java [hunk 4]
```diff
@@ -1,6 +1,753 @@
...
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    // Editing a user's info. Users can only edit their username,
+    // email or profile pictures
+    public static synchronized void editUser(UUID userId, JsonObject newUserInfo) {
```
Comment (MMDPROJECT) [inline src/main/java/Server/Database/Query.java:158]: here in the `newUserInfo`, `Service` tells us where is `profile_path` right?

File: src/main/java/Server/Database/Query.java [hunk 5]
```diff
@@ -1,6 +1,753 @@
...
+     * 2: Object: Someone or something being acted upon
+     * 3: Action: The act that's being one
+     * Type of actions:
+     * ADD_ARTIST_SOCIALMEDIA: Artist adding a social media to their page
+     * ADD_ARTIST_ALBUM: Artist adding an album to their page
+     * ALBUM_ADD_TRACK: Adding a track to an album
+     * PLAYLIST_ADD_TRACK: Adding a track to a playlist
+     * USER_FOLLOW_USER: User following a user
+     * USER_FOLLOW_ARTIST: User following an artist
+     * USER_CREAT_PLAYLIST: User creating a playlist
```
Comment (MMDPROJECT) [inline src/main/java/Server/Database/Query.java]: **CREATE**

File: src/main/java/Server/Database/Query.java [hunk 6]
```diff
@@ -1,6 +1,753 @@
...
+            
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    public static synchronized void createPlaylist(UUID userId, JsonObject playListJson) {
+        final String query = """
+            INSERT INTO Playlist values(?, ?, ?, ?, ?, ?, ?, ?);
```
Comment (MMDPROJECT) [inline src/main/java/Server/Database/Query.java]: I think here we have an extra `?`

**Review by MMDPROJECT — COMMENTED** [review]
[Here you can check my reviews](https://github.com/Navid-Ebadi-2003/spotify_project/pull/14/files/5eac743d9775f29899643fe33fde4385d82addda)

**Comment by MMDPROJECT** [comment]
I have reviewed your code, most of it is prefect except a few places that I have commented above. 
Also in **Linking backend to front** phase if we needed an extra query or something like that, I think we can make changes so easily. so feel free to merge your pull request please @Retriever5500

File: src/main/java/Server/Database/Query.java (hunk 2 above)
Comment (Retriever5500) [reply src/main/java/Server/Database/Query.java:83]: Actually I was thinking about it, wanted to ask your opinion. For now let us merge it and think about hashing it on a later PR.

File: src/main/java/Server/Database/Query.java (hunk 4 above)
Comment (Retriever5500) [reply src/main/java/Server/Database/Query.java:158]: Yeah it should. Because it's what is downloading the profile picture.

**Pull request merged**